src/polymer/content-to-slot-declarations.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import * as parse5 from 'parse5';
+import {treeAdapters} from 'parse5';
+import {Document, ParsedHtmlDocument, Severity} from 'polymer-analyzer';
+
+import {HtmlRule} from '../html/rule';
+import {registry} from '../registry';
+import {FixableWarning} from '../warning';
+
+import stripIndent = require('strip-indent');
+
+
+const p = dom5.predicates;
+
+// TODO: this should be in default collections, but it shouldn't have a
+//     fix, because the fix isn't safe, it introduces a breaking change.
+//     https://github.com/Polymer/polymer-linter/issues/111
+
+class ContentToSlotDeclarations extends HtmlRule {
+  code = 'content-to-slot-declarations';
+  description = stripIndent(`
+      Warns when using <content> instead of Shadow Dom v1's <slot> element.
+  `).trim();
+
+  async checkDocument(parsedDocument: ParsedHtmlDocument, document: Document) {
+    const warnings: FixableWarning[] = [];
+
+    this.convertDeclarations(parsedDocument, document, warnings);
+
+    return warnings;
+  }
+
+  convertDeclarations(
+      parsedDocument: ParsedHtmlDocument, document: Document,
+      warnings: FixableWarning[]) {
+    for (const element of document.getFeatures({kind: 'polymer-element'})) {
+      const domModule = element.domModule;
+      if (!domModule) {
+        continue;
+      }
+      const template = dom5.query(domModule, p.hasTagName('template'));
+      if (!template) {
+        continue;
+      }
+      const contentNodes = dom5.queryAll(
+          treeAdapters.default.getTemplateContent(template),
+          p.hasTagName('content'));
+      const slots = new Set<string>();
+      for (const contentNode of contentNodes) {
+        const warning = new FixableWarning({
+          code: 'content-to-slot-declaration',
+          message:
+              `<content> tags are part of the deprecated Shadow Dom v0 API. ` +
+              `Replace with a <slot> tag.`,
+          parsedDocument,
+          severity: Severity.WARNING,
+          sourceRange: parsedDocument.sourceRangeForStartTag(contentNode)!
+        });
+        const attrs = [...contentNode.attrs];
```

Comment:
got lost on what's happening in this block (here to line 85). Comments/split it in a separate function maybe?